tools/README.md

```diff
@@ -0,0 +1,90 @@
+# Ops Tools
+
+A collection of tools to use when developing and maintaining Ops charms.
+
+## charmcraft.yaml generation
+
+The definition of charm configuration options and actions should have a single source of truth. Our preference is that truth is in the charm code: this is in an expressive language (Python), that allows more complex type definitions and validation than possible in the Juju config schema language or action parameter JSONSchema. However, Juju and Charmcraft need to find these schema in the `charmcraft.yaml` file.
+
+The package provides tooling to generate the appropriate `charmcraft.yaml` sections for `config` and `actions` from config and action classes in the charm Python code.
+
+### Usage
+
+For example, if your charm contains a single config class in `src/charm.py` called `Config` and three actions, which all end with 'Action' (and no other classes have that name):
+
+```bash
+update-charmcraft --config-class Config --action-class '.+Action'
```

Comment:
`update-charmcraft` seems like a very generic name. Should we include "schema", like `update-charmcraft-schema`?